api/api-cache.php

```diff
@@ -0,0 +1,177 @@
+<?php
+require_once('../template/top.php');
+
+// RegEx to match certain endpoints we want to ignore, e.g., orders
+const IGNORED_ENDPOINTS = '/(?:orders)/i';
+/**
+ * Gets the first entry from the API cache that has a matching endpoint.
+ * @param string $endpoint The endpoint for the API request
+ * @return array|null The first entry that matches or null if no entry exists
+ */
+function get_cached_api_response(string $endpoint): ?array
+{
+    global $db;
+    $q = $db->query('SELECT * FROM api_cache WHERE endpoint = "' . $db->real_escape_string($endpoint) . '"');
+    if ($q) {
+        $r = $q->fetch_array(MYSQLI_ASSOC);
+        return $r;
+    }
+    return null;
+}
+
+class CacheResult{
+    /**
+     * @var string $content The response content
+     */
+    public $content;
+    /**
+     * @var bool $curl_executed true if curl_exec was called, false otherwise
+     */
+    public $curl_executed;
```

Comment:
`curl_executed` is a bad variable name, it could suggest executed is true anytime there isn't an error for example and in the future we may not stay with curl, be as verbose and specific as possible: perhaps, `fetched_new_content` or something

Reply:
Changed in c289fad. Also changed the respective parameter in the constructor